util/predicates/generic_predicates.go

```diff
@@ -157,6 +177,12 @@ func ResourceNotPaused(logger logr.Logger) predicate.Funcs {
 	}
 }
 
+// ResourceNotPausedAndHasLabel returns a predicate that returns true only if the
+// ResourceNotPaused and ResourceHasLabel predicates return true.
+func ResourceNotPausedAndHasLabel(logger logr.Logger, labelValue string) predicate.Funcs {
```

Comment:
Something similar was implemented before but then removed. I tried to follow that structure and naming.

Reply:
If there are not performance concerns, I would prefer to keep paused and filter by label as a separated predicates